python/.coveragerc

```diff
@@ -20,5 +20,5 @@ skip_empty = true
 
 [run]
 omit =
-    # omit this single file
-    src/iceberg/literals.py
+    src/iceberg/expression/literals.py
```

Comment:
nit: should we make the directory name plural, as in `expressions`? I'm thinking of how there are different types of expressions that will be defined here like And, Or, Not etc...

Reply:
ops, yes you are right @samredai , the only problem with that is that [expressions.py](https://github.com/apache/iceberg/blob/master/python/src/iceberg/expressions.py#L21) already exists so it will conflict when I try to import is as module. As you suggested, I plan to rename the file in a separate PR to get the directory naming correctly as what you suggested, what do you think?

Reply:
Ah right, I think we should rename that file to "operations.py" since it currently only contains stuff related to operations.

Reply:
Given that no other places use literals, I am wondering if we can remove those unfinished code (e.g. todos) from this PR and move them into another PR? So we can still achieve the same level of code coverage for merged code.

Reply:
@jun-he , I understand that from logical perspective, it make sense to have consistent standard where all merged codes are conforming to the same requirement on code coverage. But in this specific Literal cases, I found many places of coverage gap comes from base base such as 

- BaseLiterals
<img width="767" alt="image" src="https://user-images.githubusercontent.com/5961173/157098041-e3ded4f6-466f-4b04-b159-91b982b36819.png">

- ComparableLiterals
<img width="644" alt="image" src="https://user-images.githubusercontent.com/5961173/157098129-5d7c1d13-3349-477a-8ac6-0f87119c10f3.png">

Which if we remove the code for those, then it does not make sense for rest of the code to check in. So there's no very clean way of separating them out. What we can do here is, either ask for more code coverage for internal method of base class, to merge it within same PR, or allow for some temporary exclusion and assign the specific issue and come back in a separate PR.

Reply:
> yep I agree, do you think it would be better to do the renaming in separate PR?

Yes that sounds good to me. It's an easy follow-up PR. 👍

Reply:
re: coverage--If all of this logic in the base classes are not used anywhere, we shouldn't need them at all right? I think defining these as ABC classes will allow us to provide more clarity by clearly decorating what's an `@abstractmethod` and leaving the function body empty (just a doc string). That's a clear signal to coverage as well that there's nothing to cover. Any concrete methods in an ABC class would be there because they're used somewhere which means they can be covered in tests. Similar to what's in [io/base.py](https://github.com/apache/iceberg/blob/master/python/src/iceberg/io/base.py).